std/bitmanip.d

```diff
@@ -765,9 +764,18 @@ struct DoubleRep
 }
 
 /**
- * An array of bits.
- */
+An array of bits, where the bits are packed in `size_t` values, so the
+dimension, i.e. underlying array size, is effectively
+`dim == ceil(length / size_t.sizeof * 8)`.
 
+`BitArray` is a `RandomAccessRange` and provides all the bitwise operators
+applicable to builtin integers: `&`, `|`, `^`, `~`, as well as bunch of
+standard methods operating effectively on the bits.
+
+The methods `all`, `any`, `none`, `sort`, whose generic equivalents can be
+found in the $(REF , std, algorithm), are specialized for `BitArray` to be
+faster - they operate in complexity relative to its dimension instead of the length.
+*/
 struct BitArray
```

Comment:
This type has nothing to do with the rest of the module, and should be in its own module.